web/src/compose_ui.ts

```diff
@@ -624,6 +624,48 @@ export function handle_keyup(
     rtl.set_rtl_class_for_textarea($textarea);
 }
 
+/**
+ * True if the cursor in `$textarea` sits between an opening run of ` and its
+ * still‑missing matching closer.  Works for any run‑length (`, ```, …).
+ */
+export function cursor_inside_inline_span($textarea: JQuery<HTMLTextAreaElement>): boolean {
+    const textareaElement = $textarea[0];
+    if (!textareaElement || typeof textareaElement.selectionStart !== "number") {
+        return false;
+    }
+    // jQuery.val() can be string | number | string[] | undefined.
+    const val = $textarea.val();
+    if (typeof val !== "string") {
+        return false;
+    }
```

Comment:
I don't think it can actually be anything but a string here; you should assert rather than adding a `return` statement that can never fire.